test/metabase/query_processor/middleware/constraints_test.clj

```diff
@@ -51,3 +52,20 @@
            (add-default-userland-constraints
             {:constraints {:max-results 5, :max-results-bare-rows 10}
              :middleware  {:add-default-userland-constraints? true}})))))
+
+(deftest load-constraints-settings-from-config-test
+  (testing "loading user set constraint settings from config"
+    (mt/with-temporary-setting-values [max-results 10 max-results-bare-rows 5]
+      (is (= {:constraints {:max-results (setting/get :max-results)
+                            :max-results-bare-rows (setting/get :max-results-bare-rows)}}
+             (add-default-userland-constraints
+              {:constraints {:max-results 10
+                             :max-results-bare-rows 5}}))))))
+
+(deftest default-constraints-settings-test
+  (testing "if no user set constraint settings are found, use the default values"
+     (is (= {:constraints {:max-results (setting/get :max-results)
+                         :max-results-bare-rows (setting/get :max-results-bare-rows)}}
+          (add-default-userland-constraints
+           {:constraints {:max-results 10000
+                          :max-results-bare-rows 2000}})))))
```

Comment:
As with the test above, `add-default-userland-constraints` is always going to return the map you pass in without any changes unless it has 

```clj
{:middleware  {:add-default-userland-constraints? true}}
```

in it **and** it doesn't already have these keys, so you're basically just testing `(= x x)` here:

```clj
(add-default-userland-constraints
 {:constraints {:max-results 1 :max-results-bare-rows 2}})
;; => {:constraints {:max-results 1, :max-results-bare-rows 2}}

(add-default-userland-constraints
 {:constraints {:max-results 3 :max-results-bare-rows 4}})
;; => {:constraints {:max-results 3, :max-results-bare-rows 4}}
```

I think `add-constraints-test` covers the default situation just fine so it's ok to remove this test.

```suggestion
```